app/code/Magento/Deploy/Collector/Collector.php

```diff
@@ -93,6 +93,9 @@ public function collect()
                 if ($file->getModule() && !$this->moduleManager->isEnabled($file->getModule())) {
                     continue;
                 }
+
+                $fileName = $file->getFileName() == null ? '' : $this->fileNameResolver->resolve($file->getFileName());
+                $file->setDeployedFileName($fileName);
```

Comment:
Could you pls explain this change?
Looks like it's doing nothing. Since just after we have the `if` condition that skips empty file name and null as well.

Reply:
right,
```
              if (!$file->getFileName()) {
                  continue;
              }
```
seems to be the fix already, which is not yet there in 2.4.4-p2